### PR TITLE
Resolve symlink path for qemu directory if possible

### PR DESCRIPTION
**Pull request by n8henrie**
Fixes https://github.com/containers/podman/issues/17026
Fixes https://github.com/NixOS/nixpkgs/issues/169118

Related: https://github.com/NixOS/nixpkgs/pull/163015

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by n8henrie** [comment]
> Please write a regression test for what you're fixing. Even if it
seems trivial or obvious, try to add a test that will prevent
regressions.

Adding `[NO NEW TESTS NEEDED]` as I don't see any related existing tests, the only change is following symlinks on darwin (not fedora).

However, an analogous change *may* be useful for nix users of aarch64-linux -- I don't know, will experiment.

File: pkg/machine/qemu/options_darwin_arm64.go
Comment (rhatdan) [inline pkg/machine/qemu/options_darwin_arm64.go:61]: This should probably be done in the containers/common cfg.FineHelperBinary() so that all callers get the benefit.

Comment (n8henrie) [reply pkg/machine/qemu/options_darwin_arm64.go:61]: Thanks for your input!

I'd be happy to make a PR there, though I hesitate to apply it so broadly, as resolving the symlink could change behavior in some cases; I thought it was safe and reasonable in this case as we're looking specifically for the `qemu-system-aarch64` binary, and only on aarch64-darwin, and this seems to solve a semi-common issue for nix + podman users on this specific platform.

For example, resolving the symlink could change the returned filename in unexpected ways (which I think is unlikely to be a problem with `qemu-system-aarch64`, but could in other contexts).

```go
package main

import (
	"os"
	"path/filepath"
)

func main() {
	cwd, _ := os.Getwd()
	path, name := cwd, "fake-link"
	fullpath := filepath.Join(path, name)
	if fi, err := os.Stat(fullpath); err == nil && fi.Mode().IsRegular() {
		println("no error", fullpath)

		if p, err := filepath.EvalSymlinks(fullpath); err == nil {
			println("resolved", p)
			return
		}
		return
	}
	println("whups")
}
```

```console
$ ln -s $(which qemu-system-aarch64) ./fake-link
$ go run .
no error /var/folders/kb/tw_lp_xd2_bbv0hqk4m0bvt80000gn/T/tmp.sttNLyt8Xi/fake-link
resolved /nix/store/b5msl7ib515wdfvgvcxl370x1p1p7s4r-qemu-7.2.0/bin/qemu-system-aarch64
```

Comment (vrothberg) [reply pkg/machine/qemu/options_darwin_arm64.go:61]: @rhatdan PTAL

**Comment by n8henrie** [comment]
@rhatdan -- it seems that @Luap99 [thinks this PR belongs in podman](https://github.com/containers/common/pull/1303#issuecomment-1408390502), and I think it makes sense here as well (at least for now -- going forward, if similar issues pop up for other projects or architectures it could be adopted for them as well).

As shown [here](https://github.com/containers/common/pull/1303#issuecomment-1407596059), this patch seems to work well and would make a big difference for podman users on aarch64-darwin that are using nix (as opposed to homebrew) for package management, which I think is a growing group.

I'm not sure what's going on with those test failures though. My only code change was in `pkg/machine/qemu/options_darwin_arm64.go`, so I have to assume the windows failure is unrelated.

**Comment by rhatdan** [comment]
This is fine with me.

**Comment by n8henrie** [comment]
@rhatdan thanks for your help and feedback with this issue. It looks like you're in the list of approvers. Is this blocked on anything else from my side?

It looks like the CI failure is probably unrelated:

```
Error: copying system image from manifest list: ... dial tcp: lookup cdn03.quay.io: no such host
```

**Comment by Luap99** [comment]
@n8henrie Can you please rebase? While it is not required we often have issues with older PRs that where not fully rebased on main.

I will merge it then.

**Comment by n8henrie** [comment]
@Luap99 done, thanks!

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**
